cli/injector/main.go

```diff
@@ -337,36 +430,18 @@ func injectAndWait(cmd *cobra.Command, args []string) {
 
 // cleanAndExit cleans the disruption with the configured injector and exits nicely
 func cleanAndExit(cmd *cobra.Command, args []string) {
-	errs := []error{}
-
-	for _, inj := range injectors {
-		// start cleanup which is retried up to 3 times using an exponential backoff algorithm
-		if err := backoff.RetryNotify(inj.Clean, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), 3), retryNotifyHandler); err != nil {
-			handleMetricError(ms.MetricCleaned(false, cmd.Name(), nil))
-
-			errs = append(errs, err)
-		}
-
-		handleMetricError(ms.MetricCleaned(true, cmd.Name(), nil))
-	}
+	nbErrorsOnClean, reason := clean(cmd.Name(), true)
 
 	// 1 or more injectors failed to clean, log and fatal
-	if len(errs) != 0 {
-		var combined strings.Builder
-
-		for _, err := range errs {
-			combined.WriteString(err.Error())
-			combined.WriteString(",")
-		}
-
-		log.Fatalw(fmt.Sprintf("disruption cleanup failed on %d injectors (comma separated errors)", len(errs)), "errors", combined.String())
+	if nbErrorsOnClean > 0 {
```

Comment:
Same thing here, I think that using a simple boolean in the clean function return is enough since we already log errors in that function. We can keep the last fatal log that will exit (non 0) the injector if cleanup failed so we can catch that later.